plugins/plugin-webpack/plugin.js

```diff
@@ -161,7 +172,22 @@ module.exports = function plugin(config, args) {
   if (!cssOutputPattern.endsWith(".css")) {
     throw new Error("Output Pattern for CSS must end in .css");
   }
-  
+
+  // Default options for HTMLMinifier
+  // https://github.com/kangax/html-minifier#options-quick-reference
+  let htmlMinifierOptions = {
+    collapseWhitespace: true,
+    removeComments: true,
+    removeEmptyAttributes: true,
+    removeRedundantAttributes: true,
+    removeScriptTypeAttributes: true,
+    removeStyleLinkTypeAttributes: true,
+  };
+
+  if (typeof args.htmlMinifierOptions !== 'undefined') {
+    htmlMinifierOptions = args.htmlMinifierOptions;
+  }
```

Comment:
I think as a user I would expect my passed options to be merged with the defaults. With that implementation, if I just set `collapseWhitespace: false`, then `htmlMinifierOptions` will result in `{collapseWhitespace:false}` instead of `{ collapseWhitespace: false, removeComments: true, removeEmptyAttributes: true, removeRedundantAttributes: true, removeScriptTypeAttributes: true, removeStyleLinkTypeAttributes: true }`

I would suggest something like this

```suggestion
  const defaultHtmlMinifierOptions = {
    collapseWhitespace: true,
    removeComments: true,
    removeEmptyAttributes: true,
    removeRedundantAttributes: true,
    removeScriptTypeAttributes: true,
    removeStyleLinkTypeAttributes: true,
  };

  const htmlMinifierOptions = Object.assign({}, defaultHtmlMinifierOptions, args.htmlMinifierOptions)
```

Reply:
+1. As @gr2m said, this plugin should set defaults (the ones you have are great), which the user can override. But if a user provides one option, it shouldn’t blow away all the other defaults set.

Reply:
But what if I only want to have one option or avoid something which is part of the defaults? This is not possible when I merge them together.

Reply:
what do you mean with "avoid something which is part of the defaults"? Could you share a use case?

Reply:
For example when I want to have the default configuration except for `removeStyleLinkTypeAttributes`.

I don't have a specific use case in mind, I just don't like the idea of forcing everyone to use the defaults.

Reply:
in that case you just set `removeStyleLinkTypeAttributes: false`, and all other options will default to their respective defaults. With your current implementations, all other options would fall back to `undefined`

Reply:
Ok, fair point.

What is if I provide `false` to disable the minification completely?

Reply:
Ah, Good Point, you have to check for that before the merge

Reply:
I've updated my change suggestion to account for that